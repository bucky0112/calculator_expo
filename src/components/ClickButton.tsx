import {
  Text,
  TouchableOpacity,
  TextProps,
  TouchableOpacityProps,
} from 'react-native'
import classNames from 'classnames'

type ClickButtonProps = TouchableOpacityProps & {
  text: string
  size?: 'double'
  theme?: 'primary' | 'secondary' | 'accent'
  textProps?: TextProps
}

const ClickButton = ({
  onPress,
  text,
  size,
  theme = 'primary',
  textProps,
  ...rest
}: ClickButtonProps) => {
  

  return (
    <TouchableOpacity
      onPress={onPress}
      className={classNames({
        "rounded-full flex-1 items-center justify-center m-2": true,
        "h-[84px]": true,
        "w-[84px]": true,
        'w-full': size === 'double',
        'bg-primary-btn': theme === 'primary',
        'bg-secondary-btn': theme === 'secondary',
        'bg-accent-btn': theme === 'accent'
      })}
      {...rest}
      testID='click-button'
    >
      <Text
        className={classNames({
          'text-3xl font-bold text-center': true,
          'text-white': theme === 'secondary' || theme === 'primary',
        })}
        {...textProps}
      >
        {text}
      </Text>
    </TouchableOpacity>
  )
}

export { ClickButton }
