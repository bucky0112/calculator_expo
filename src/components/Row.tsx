import { View } from 'react-native'

type RowProps = {
  children: React.ReactNode;
};

const Row = ({ children, ...rest }: RowProps) => {
  return (
    <View className='flex-row items-center gap-1' {...rest}>
      {children}
    </View>
  )
}

export { Row }
