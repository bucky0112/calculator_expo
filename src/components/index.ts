export { ClickButton } from './ClickButton'
export { Row } from './Row'
