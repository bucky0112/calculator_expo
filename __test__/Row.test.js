import { Row } from "../src/components/Row"
import { render } from "@testing-library/react-native"

describe("Row", () => {
  it("renders correctly", () => {
    const { toJSON } = render(<Row />)
    expect(toJSON()).toMatchSnapshot()
  })
})
