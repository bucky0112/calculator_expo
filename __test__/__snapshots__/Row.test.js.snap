// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "alignItems": "center",
      },
      Object {
        "marginLeft": -4,
        "marginTop": -4,
      },
    ]
  }
/>
`;
